app/controllers/api/order_cycles_controller.rb

```diff
@@ -1,11 +1,18 @@
 module Api
   class OrderCyclesController < Api::BaseController
     include EnterprisesHelper
-    respond_to :json
+    include ApiActionCaching
 
     skip_authorization_check
+    skip_before_filter :authenticate_user, :ensure_api_key, only: [:taxons, :properties]
+
+    caches_action :taxons, :properties,
+                  expires_in: CacheService::FILTERS_EXPIRY,
+                  cache_path: proc { |controller| controller.request.url }
```

Comment:
Do I understand correctly that this action caching is not related to the cache_service below? If this is correct, maybe we need to move this to a new PR.

I think this looks a promising form of API caching!

I am not sure I am missing something but can I approve this part of the PR and reject what's happening on the injectiondata side of this PR? If we move the injection_data methods to API endpoints we can than use this technique to cache them?

anyway, great work @Matt-Yorkley :muscle:

Reply:
I think in some cases it might be detrimental (performance-wise) to move all of the injected data to endpoints. We inject 7 different sets of AMS data in the Darkswarm layout for example (and more on specific pages). If we add an extra 7 requests per page load, we would be hitting the stack an extra 7 times on every page. The API controllers are thinner than the standard controllers, but not that thin. So maybe performance could get worse in that scenario..?

Reply:
There are some bits of injection data that would be really great to extract to the API, but would currently take a large amount of work (like the /shops page).

Reply:
I think we should move all of them to the API and then optimize the API to make it faster. I think it's very important we rely on API based solutions internally so that we can grow our public API and it's performance!

I wonder how difficult it would be. The payload would be exactly the same, we would only have to replace the angular code (it's now a variable with the name of the injected data in the controllers dependencies list) with a resource that fetches data from the API as well as creating API endpoints that will be equal to the injectos. I think we have to make this way: all these would be become public API endpoints.

Reply:
I agree with your view @luisramos0 but I understand your point @Matt-Yorkley . That's a potential downside BUT if we consider the whole picture (browser rendering, ajax calls, server latency) we might even get quicker overall browser response time if we intelligently fetch things asynchronously and don't block the page rendering. You know, there's a lot of science in there.

Reply:
On the `/shops` page specifically we have a lot of Geocoding functionality that happens in javascript libraries as part of the search. Moving all the logic to the backend would be tricky. We would also then have a lot of different combinations of search params for the endpoint which would make it unfeasible to cache the output nicely. I'd still love to do that at some point though. It's desperately needed but it's not a small job.

Reply:
I generally agree on moving things to the API and then making it faster though! I just don't think that will happen very soon.